### DISCUSSION TITLE
Default RLS Policies.

### DISCUSSION THREAD

**Discussion by The-Code-Monkey** [supabase/supabase#11792] [Feature Requests]
# Feature request

## Is your feature request related to a problem? Please describe.

I am setting up a CMS which automatically picks up new tables from the public schema and allows the user to edit the entries.
But my problem is that I have to create the same 4 RLS policies for each table.

## Describe the solution you'd like

A way to specify default RLS policies for all tables, but these will be overwritten by Table specific RLS policies

## Describe alternatives you've considered

There isn't really any alternatives apart from doing this manually every time I want to add a table. I guess maybe a clone button on RLS policies so you can copy them over easier.

## Additional context

If you want any more info or access to the CMS that im building to see what i mean let me know.

**Comment by steve-chavez** [comment]
> But my problem is that I have to create the same 4 RLS policies for each table.

Can you share a sample of those? If there's a convincing case we could try to propose this enhancement to the PostgreSQL dev team.

**Comment by The-Code-Monkey** [comment]
<img width="1242" alt="Screenshot 2023-01-12 at 08 13 20" src="https://user-images.githubusercontent.com/31218527/212013155-f9678273-c381-4cbd-8ae2-5559026233fc.png">

I have to put these same 4 policies on each table, there isnt anything super special about them apart from they check if the user is logged in via the supabase auth. 

There are a couple of tables where they would be overridden in my case as i use an extra auth code layer on top of the auth to make it extra secure when signing up. but that would be overridden when adding a specific set of rules to that table. 

So what im asking for is 

A way to set a default set of policies to all tables
Unless there is a policy added specifically to that table then those would apply instead of the defaults.

**Comment by kiwicopple** [comment]
We discussed this internally and are considering showing "existing" RLS policies for other tables in the RLS editor. We'll work on some RLS improvements over the next month. Thanks for the feedback @The-Code-Monkey 

I'll convert this one to a discussion for now (in the Feautre Requests category)

**Reply by The-Code-Monkey** [reply]
Ooh thats a better idea. Thanks for discussing it. I await to see the progress in the future with other things.

**Reply by vipulb2** [reply]
This could be a good first task for someone starting new.👍

**Reply by tobysimone** [reply]
Is there an update on this? I'm going through the process of applying the same couple policies to some new tables and it's a painful/slow process.
